### PR TITLE
bug: Fix fundAccount always waiting for Indexer

### PR DESCRIPTION
**Pull request by jmintuitive**
### Description

Funding an account using the faucet currently always triggers the logic to "waitForIndexer" since `undefined !== false` is `true`.
image
This PR updates the logic so that it only waits for the indexer if `options.waitForIndexer` is `true`. 

This was found by trying to run `simple_transfer.ts` in the example folder, and immediately getting this error: 
```
 [AptosApiError]: Indexer error: field 'processor_status' not found in type: 'query_root'
    at Gn (/Users/jacksonmills/aptos-ts-sdk/dist/common/index.js:1:36526)
    at processTicksAndRejections (node:internal/process/task_queues:95:5)
    at async f (/Users/jacksonmills/aptos-ts-sdk/dist/common/index.js:470:383)
    at async tc (/Users/jacksonmills/aptos-ts-sdk/dist/common/index.js:470:571)
    at async Hn (/Users/jacksonmills/aptos-ts-sdk/dist/common/index.js:470:694)
    at async Zn (/Users/jacksonmills/aptos-ts-sdk/dist/common/index.js:470:12500)
    at async gn.fundAccount (/Users/jacksonmills/aptos-ts-sdk/dist/common/index.js:470:55884)
    at async example (/Users/jacksonmills/aptos-ts-sdk/examples/typescript/simple_transfer.ts:57:24) {
  url: 'https://api.devnet.aptoslabs.com/v1/graphql',
  status: 200,
  statusText: 'OK',
  data: { errors: [ [Object] ] },
  request: {
    url: 'https://api.devnet.aptoslabs.com/v1/graphql',
    method: 'POST',
    originMethod: 'getProcessorStatuses',
    path: '',
    body: {
      query: '\n' +
        '    query getProcessorStatus($where_condition: processor_status_bool_exp) {\n' +
        '  processor_status(where: $where_condition) {\n' +
        '    last_success_version\n' +
        '    processor\n' +
        '    last_updated\n' +
        '  }\n' +
        '}\n' +
        '    '
    },
    contentType: undefined,
    acceptType: undefined,
    params: undefined,
    overrides: { WITH_CREDENTIALS: false, HEADERS: {} }
  }
}
```

While this fix solves the immediate issue for the tutorial, it does NOT address why `processor_status` was not found in the `query_root` for the devnet indexer. **That should be investigated separately.**

(I also updated the contributing guide to clearly spell out how to test code since it was very easy to miss when skimming.)

### Test Plan

BEFORE:
Screenshot

AFTER:
image

### CODE REVIEW

**Review by gregnazario — APPROVED** [review]

File: CONTRIBUTING.md [hunk 1]
```diff
@@ -26,11 +26,20 @@ If you know Markdown, you can already contribute!
...
 
-* If you have added code that should be tested, add unit tests.
-* If you have changed APIs, update the documentation. Make sure the documentation builds.
-* Ensure all formatting applies with `pnpm fmt`.
-* Ensure all tests and lints pass on each and every commit that is part of your pull request using `pnpm test && pnpm lint`.
-* Submit your pull request.
+1. If you have added code that should be tested, add unit tests.
+2. If you have changed APIs, update the documentation. Make sure the documentation builds.
+3. Ensure all formatting applies with `pnpm fmt`.
+4. Ensure all tests and lints pass on each and every commit that is part of your pull request using `pnpm test && pnpm lint`.
```
Comment (0xmaayan) [inline CONTRIBUTING.md:32]: make sure to update CHANGELOG =)

**Review by banool — APPROVED** [review]

**Comment by 0xmaayan** [comment]
ok, so tests are failing because of this change.

**Comment by gregnazario** [comment]
> ok, so tests are failing because of this change.

We'll need to probably set the field for a lot of tests

**Review by 0xmaayan — CHANGES_REQUESTED** [review]
we should not change the default fundAccount behavior and keep it to always wait for indexer. lets revert this change and keep all the other stuff

**Comment by jmintuitive** [comment]
@0xmaayan Updated the logic so it's more explicit that `undefined` means that we wait for the indexer (this is functionally equivalent to the old default of always waiting :) )

This is ready for re-review when you have a chance :)

File: examples/typescript/your_fungible_asset.ts [hunk 2]
```diff
@@ -146,7 +146,11 @@ async function main() {
   console.log(`Charlie: ${charlie.accountAddress.toString()}`);
 
   await aptos.fundAccount({ accountAddress: alice.accountAddress, amount: 100_000_000 });
-  await aptos.fundAccount({ accountAddress: bob.accountAddress, amount: 100_000_000 });
+  await aptos.fundAccount({
+    accountAddress: bob.accountAddress,
+    amount: 100_000_000,
+    options: { waitForIndexer: true },
```
Comment (0xmaayan) [inline examples/typescript/your_fungible_asset.ts:152]: dont need it

File: examples/typescript/your_coin.ts [hunk 3]
```diff
@@ -97,7 +97,11 @@ async function main() {
   console.log(`Bob: ${bob.accountAddress.toString()}`);
 
   // Fund alice account
-  await aptos.fundAccount({ accountAddress: alice.accountAddress, amount: 100_000_000 });
+  await aptos.fundAccount({
+    accountAddress: alice.accountAddress,
+    amount: 100_000_000,
+    options: { waitForIndexer: true },
```
Comment (0xmaayan) [inline examples/typescript/your_coin.ts:103]: dont need it

File: examples/typescript/simple_sponsored_transaction.ts [hunk 4]
```diff
@@ -41,7 +41,11 @@ const example = async () => {
 
   // Fund Alice and sponsor accounts
   await aptos.fundAccount({ accountAddress: aliceAddress, amount: ALICE_INITIAL_BALANCE });
-  await aptos.fundAccount({ accountAddress: sponsorAddress, amount: SPONSOR_INITIAL_BALANCE });
+  await aptos.fundAccount({
+    accountAddress: sponsorAddress,
+    amount: SPONSOR_INITIAL_BALANCE,
+    options: { waitForIndexer: true },
```
Comment (0xmaayan) [inline examples/typescript/simple_sponsored_transaction.ts:47]: dont need it

File: examples/javascript/simple_sponsored_transaction.js [hunk 5]
```diff
@@ -37,7 +37,11 @@ const example = async () => {
 
   // Fund Alice and sponsor accounts
   await aptos.fundAccount({ accountAddress: aliceAddress, amount: ALICE_INITIAL_BALANCE });
-  await aptos.fundAccount({ accountAddress: sponsorAddress, amount: SPONSOR_INITIAL_BALANCE });
+  await aptos.fundAccount({
+    accountAddress: sponsorAddress,
+    amount: SPONSOR_INITIAL_BALANCE,
+    options: { waitForIndexer: true },
```
Comment (0xmaayan) [inline examples/javascript/simple_sponsored_transaction.js:43]: dont need it

**Comment by jmintuitive** [comment]
@0xmaayan Updated to remove the unnecessary `option: { waitForIndexer = true }` lines :)

**Review by 0xmaayan — APPROVED** [review]
LGTM. please update CHANGELOG with the check changes in fundAccount

**Pull request merged**